### PR TITLE
feat(connect accounts): #62 Modified user model and added functionality to store secondary user

### PR DESCRIPTION
**Pull request by VenkatVenkatesh**
- Added a boolean `isPrimary` in user model
- Added `socialConnections` model and used as type definition for `socialConnections` property in user model
- Implemented the functionality to add connected account details with `isPrimary` as false in user collection.
- Modified the existing user save functionality in login component.

### CODE REVIEW

File: web/src/app/models/user.model.ts [hunk 1]
```diff
@@ -1,15 +1,19 @@
+import { SocialConnections } from './socialConnections';
+
 export interface User {
   displayName: string;
   photoUrl: string;
   uid: string;
   creationAt: string;
   lastSignInAt: string;
+  isPrimary: boolean;
```
Comment (eddiejaoude) [inline web/src/app/models/user.model.ts:9]: Is this needed? The user will always be primary because the connected socials are secondary.

Comment (VenkatVenkatesh) [reply web/src/app/models/user.model.ts:9]: @eddiejaoude When we start handling #65 , this field would be easy check.

Comment (eddiejaoude) [reply web/src/app/models/user.model.ts:9]: I think this adds code with no benefit, as checking a small collection is simple also. If you prefer leave it in for now and we can remove it later on

Comment (VenkatVenkatesh) [reply web/src/app/models/user.model.ts:9]: Okay @eddiejaoude for now we will leave it, once I start working on #65 and #66 , will get a clear idea about this.

**Review by eddiejaoude — APPROVED** [review]

**Pull request merged**